rust-connectors/sources/mqtt/src/main.rs

```diff
@@ -93,9 +93,26 @@ fn main() -> Result<(), MqttConnectorError> {
                 %client_id
             );
         }
-        let mut mqttoptions = MqttOptions::try_from(url)?;
+        let mut mqttoptions = MqttOptions::try_from(url.clone())?;
         mqttoptions.set_keep_alive(mqtt_timeout_seconds);
+        if url.scheme() == "mqtts" {
+            let mut root_cert_store = rustls::RootCertStore::empty();
+            for cert in
+                rustls_native_certs::load_native_certs().expect("could not load platform certs")
+            {
+                root_cert_store
+                    .add(&rustls::Certificate(cert.0))
+                    .expect("Failed to parse DER");
+            }
+            let client_config = ClientConfig::builder()
+                .with_safe_defaults()
+                .with_root_certificates(root_cert_store)
+                .with_no_client_auth();
+
+            mqttoptions.set_transport(Transport::tls_with_config(client_config.into()));
+        }
         let producer = opts.common.create_producer().await?;
+        info!("MQTT OPTIONS {:?}", mqttoptions);
```

Comment:
Should we remove this line, right? or maybe change to `debug!`

Reply:
I didn't mean to leave it in.